content/blog/2025-05-08-go-client-GA.md

```diff
@@ -0,0 +1,503 @@
+
+### Authentication and TLS
+
+Secure your connections with built-in authentication and TLS support:
+
+```go
+// Configure with authentication
+config := api.NewGlideClientConfiguration().
+    WithAddress(&api.NodeAddress{Host: "localhost", Port: 6379}).
+    WithCredentials(api.NewServerCredentials("username", "password")).
+    WithUseTLS(true) // Enable TLS for encrypted connections
+```
+
+### Request Timeout and Handling
+
+Fine-tune timeout settings for different workloads:
+
+```go
+// Set a longer timeout for operations that may take more time
+config := api.NewGlideClientConfiguration().
+    WithAddress(&api.NodeAddress{Host: "localhost", Port: 6379}).
+    WithRequestTimeout(500) // 500ms timeout
+```
+
+## Behind the Scenes: Technical Architecture
+
+The Valkey GLIDE Go client is built on top of the Valkey GLIDE core. The core framework is written in Rust (lib.rs), which exposes public functions. These functions are converted to a C header file using Cbindgen. The Go client then uses CGO to call these C functions, providing Go developers with an idiomatic interface while leveraging Rust's performance advantages. This architecture ensures consistent behavior across all Valkey GLIDE language implementations (Java, Python, Node.js, and Go) while maintaining performance and reliability.
+
+### Component details
+
+```text
++------------+      +------+      +------------+      +------------+      +------------+
+|            |      |      |      |            |      |            |      |            |
+|    Go      |----->|      |----->|  C Header  |----->|    Rust    |----->|   Valkey   |
+|  Client    |      |  CGO |      |  cbindgen  |      |    Core    |      |   Server   |
+|            |<-----|      |<-----|            |<-----|            |<-----|            |
+|            |      |      |      |            |      |            |      |            |
++------------+      +------+      +------------+      +------------+      +------------+
+```
+
+- **Go Client**: The language-specific interface for Go developers
+- **CGO**: Allows Go code to call C functions
+- **Cbindgen**: Automates the generation of C header files from Rust public APIs
+- **Rust Core**: High-performance framework that connects to and communicates with Valkey servers
+- **Rust FFI Library**: Enables cross-language function calls between Rust and other languages
+
+## Transaction Support
+
+The Go client now includes full transaction support, enabling atomic operations across multiple commands.
+
+### Transaction Lifecycle
+
+**Create Transaction:**
+
+ Initialize a transaction object.
+Use `api.NewTransaction(client)` to create a new transaction with established client as an argument
+`cmd := tx.GlideClient` grants access to a transaction-scoped commands interface, allowing you to construct a sequence of operations to be executed. We can access transaction commands via the embedded client.
```

Comment:
'embedded client' raises an eyebrow for me - it leads me to different road that what I think you mean. 

Do you mean an embedded client construct (context? object? not sure if the right golang terminology)